tooling-cpu-load/src/main/java/org/jboss/eap/qe/microprofile/tooling/cpu/load/HighCPUUtils.java

```diff
@@ -0,0 +1,98 @@
+     * <p>
+     * This method binds process with Wildfly/EAP to single CPU core (to 1st core) and then causes 100% load on this core so
+     * Wildfly/EAP process has very little CPU time (almost not scheduled) on this core.
+     *
+     * @param containerProcessId process id (pid) of Wildfly/EAP process
+     * @param durationOfLoad how long will be CPU with EAP/Wildfly server under 100% load
+     * @return return process of CPU load generator
+     * @throws Exception if causing CPU load on given process fails
+     */
+    public static Process causeMaximumCPULoadOnContainer(int containerProcessId, Duration durationOfLoad)
+            throws Exception {
+        return causeMaximumCPULoadOnContainer(containerProcessId, "0", durationOfLoad);
+    }
+
+    /**
+     * Simulates that EAP/Wildfly server will be under 100% CPU load.
+     * <p>
+     * This method binds process with Wildfly/EAP to given CPU cores and then causes 100% load on those cores so Wildfly/EAP
+     * process has very little CPU time (almost not scheduled) on those cores.
+     * <p>
+     * WARNING: Remember that if all CPU cores will be specified then it might crash the test when under 100% CPU cores
+     * thus it's recommended to consume just single core.
+     *
+     * @param containerProcessId process id (pid) of Wildfly/EAP process
+     * @param cpuIds specifies CPU cores to which Wildfly/EAP server which has <code>containerProcessId</code> and cpu load
+     *        generator will be bound (for example "0-3", "0,1", ...)
+     * @param durationOfLoad how long will be CPU with EAP/Wildfly server under 100% load
+     * @return return process of CPU load generator
+     * @throws Exception if causing CPU load on given process fails
+     */
+    protected static Process causeMaximumCPULoadOnContainer(int containerProcessId, String cpuIds, Duration durationOfLoad)
+            throws Exception {
+
+        if (!LinuxProcessUtils.isLinuxEnvironment()) {
+            throw new RuntimeException("This tooling can be used only on Linux as it depends " +
+                    "OS native commands like taskset and renice.");
+        }
+        // bind container process to given CPU cores
+        LinuxProcessUtils.bindProcessToCPU(containerProcessId, cpuIds);
+        LinuxProcessUtils.setPriorityToProcess(containerProcessId,
+                LinuxProcessUtils.LOWEST_CPU_PROCESS_PRIORITY);
+
+        // start CPU load generator and bind to the same CPU cores
+        Process cpuLoadProcess = generateLoadInSeparateProcess(durationOfLoad);
+        int cpuLoadProcessId = LinuxProcessUtils.getProcessId(cpuLoadProcess);
+        LinuxProcessUtils.bindProcessToCPU(cpuLoadProcessId, cpuIds);
+        // lower CPU priority for CPU load generator so in case of 1 CPU core, test suite is still scheduled
+        LinuxProcessUtils.setPriorityToProcess(cpuLoadProcessId,
+                LinuxProcessUtils.NORMAL_CPU_PROCESS_PRIORITY + 5);
```

Comment:
I suggested to move this magic number to its own variable.

Reply:
I've removed code setting priority to process. It caused more issues then benefits. Main issue is that once EAP/WF priority is lowered then it cannot be returned back for other tests.